xbmc/peripherals/bus/virtual/PeripheralBusAddon.cpp

```diff
@@ -332,9 +332,11 @@ void CPeripheralBusAddon::GetDirectory(const std::string &strPath, CFileItemList
 
 void CPeripheralBusAddon::OnEvent(const ADDON::AddonEvent& event)
 {
-  if (typeid(event) == typeid(ADDON::AddonEvents::Enabled) ||
+  if (typeid(event) == typeid(ADDON::AddonEvents::Initilized) ||
+      typeid(event) == typeid(ADDON::AddonEvents::Enabled) ||
       typeid(event) == typeid(ADDON::AddonEvents::Disabled) ||
-      typeid(event) == typeid(ADDON::AddonEvents::InstalledChanged))
+      typeid(event) == typeid(ADDON::AddonEvents::ReInstalled) ||
+      typeid(event) == typeid(ADDON::AddonEvents::UnInstalled))
     UpdateAddons();
```

Comment:
Add also a `if (CServiceBroker::GetAddonMgr().HasType(event.id, ADDON_PERIPHERALDLL))` check.

Reply:
Sure, will add the same here